Sources/Rendering/WebGPU/Types/index.js

```diff
@@ -0,0 +1,74 @@
+// ----------------------------------------------------------------------------
+// vtkWebGPUDevice static functions
+//
+// WebGPU uses types in a many places and calls, and often those types
+// need to be associated with byte sizes, alignments, native arrays etc.
+// The folowing methods are designed to help vtk.js introspect those types.
+// WebGPU currently tends to use multiple type formats:
+//  - buffer types such as float32x4
+//  - shader types suchs as vec4<f32>
+//  - texture types such as rgba32float
+// ----------------------------------------------------------------------------
+
+function getByteStrideFromBufferFormat(format) {
+  if (!format) return 0;
+  let numComp = 1;
+  if (format.substring(format.length - 2) === 'x4') numComp = 4;
+  if (format.substring(format.length - 2) === 'x3') numComp = 3;
+  if (format.substring(format.length - 2) === 'x2') numComp = 2;
+
+  let typeSize = 4;
+  if (numComp > 1) {
+    if (format.substring(format.length - 3, format.length - 1) === '8x') {
+      typeSize = 1;
+    }
+    if (format.substring(format.length - 4, format.length - 1) === '16x') {
+      typeSize = 2;
+    }
+  } else {
+    if (format.substring(format.length - 1) === '8') typeSize = 1;
+    if (format.substring(format.length - 2) === '16') typeSize = 2;
+  }
+  return numComp * typeSize;
+}
+
+function getNativeTypeFromBufferFormat(format) {
+  if (!format) return null;
+  if (format.substring(0, 7) === 'float32') return 'Float32Array';
+  if (format.substring(0, 6) === 'snorm8') return 'Int8Array';
+  if (format.substring(0, 6) === 'unorm8') return 'Uint8Array';
+  return '';
```

Comment:
Should those test be done like `if (format.indexOf('float32') === 0) return 'Float32Array';` ? 
That would prevent any object creation.

Reply:
unless the format string could be long hence the substring?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith could be a better match

Reply:
Reworked to support the current full list of types.